src/ManagedArray.inl

```diff
@@ -63,7 +63,7 @@ CHAI_INLINE
 CHAI_HOST_DEVICE ManagedArray<T>::ManagedArray(
     uint elems, ExecutionSpace space):
   m_active_pointer(nullptr),
-  m_resource_manager(ArrayManager::getInstance()),
```

Comment:
I've been bitten by this before - this was how I ended up with the map find() call failing....

Reply:
yeah, I actually think I ran into something analogous in my development, but had forgotten about it